apps/src/templates/teacherDashboard/AssignmentVersionSelector.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { Component, PropTypes } from 'react';
+import i18n from '@cdo/locale';
+import {assignmentVersionShape} from './shapes';
+import PopUpMenu from "../../lib/ui/PopUpMenu";
+import AssignmentVersionMenuItem from './AssignmentVersionMenuItem';
+
+const styles = {
+  version: {
+    display: 'inline-block',
+    marginTop: 4,
+  },
+  dropdownLabel: {
+    fontFamily: '"Gotham 5r", sans-serif',
+  },
+  popUpMenuStyle: {
+    // must appear in front of .modal from application.scss
+    zIndex: 1051,
+  }
+};
+
+export default class AssignmentVersionSelector extends Component {
+  static propTypes = {
+    dropdownStyle: PropTypes.object,
+    onChangeVersion: PropTypes.func.isRequired,
+    selectedVersion: assignmentVersionShape,
+    versions: PropTypes.arrayOf(assignmentVersionShape),
+    disabled: PropTypes.bool,
+    showVersionMenu: PropTypes.bool,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isMenuOpen: false,
+      canMenuOpen: true,
+      targetPoint: {top: 0, left: 0},
+    };
```

Comment:
Since initial state is not dependent on props, could use [class-property syntax](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/) instead of a constructor.